spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/service/impl/DefaultTaskJobService.java

```diff
@@ -190,14 +192,45 @@ public void restartJobExecution(long jobExecutionId) throws NoSuchJobExecutionEx
 			deploymentProperties.put(DefaultTaskExecutionService.TASK_PLATFORM_NAME, platformName);
 			String taskAppName = taskJobExecution.getJobExecution().getJobParameters().getString("-spring.cloud.data.flow.taskappname");
 			taskExecutionService.executeTask(taskDefinition.getName(), deploymentProperties,
-					taskExecution.getArguments(), taskAppName);
+					restartExecutionArgs(taskExecution.getArguments(),
```

Comment:
The only caveat with this approach is that we persist the Job parameters as part of the task execution arguments (as soon as the failed task gets re-started for the first time) which I think is ok.